### PR TITLE
feat: allow specifying HTTPS_PROXY, NO_PROXY, TS_TAGS

### PR DESCRIPTION
**Pull request by jta**
This commit adds:

- the ability to optionally configure an OAuth tailscale client to connect to an existing tailnet
- the ability to specify HTTPS_PROXY and NO_PROXY for use in `terraform plan` and `apply`

Combining both features a user can proxy requests to the Observe API through an access node in a tailscale network.

### CODE REVIEW

File: .github/workflows/tf-account.yaml [hunk 1]
```diff
@@ -50,6 +72,8 @@ jobs:
 
       - name: Terraform Init
         id: init
+        # We avoid proxying on init, otherwise the exclusion
```
Comment (bendrucker) [inline .github/workflows/tf-account.yaml:75]: This should be configurable and probably enabled by default, no? If you're proxying it seems like you should be proxying all traffic by default.

File: .github/workflows/tf-account.yaml [hunk 2]
```diff
@@ -12,12 +12,25 @@ on:
...
     inputs:
       terraform-version:
         description: 'Terraform version'
         required: false
         type: string
         default: '1.5.4'
+      TS_TAGS:
+        required: false
+        type: string
+      HTTPS_PROXY:
```
Comment (bendrucker) [inline .github/workflows/tf-account.yaml:28]: Is this not related to Tailscale? I thought that the idea is essentially that the Tailscale connection is essentially providing the proxy.

File: .github/workflows/tf-account.yaml [hunk 3]
```diff
@@ -12,12 +12,25 @@ on:
...
+        required: false
+      TS_OAUTH_SECRET:
+        required: false
     inputs:
       terraform-version:
         description: 'Terraform version'
         required: false
         type: string
         default: '1.5.4'
+      TS_TAGS:
```
Comment (bendrucker) [inline .github/workflows/tf-account.yaml:25]: Inputs should be `dash-case`.

Comment (bendrucker) [reply .github/workflows/tf-account.yaml:25]: More importantly, inputs are an abstraction layer, so avoid abbreviations:

`TS_TAGS` -> `tailscale-tags`

File: .github/workflows/tf-account.yaml (hunk 2 above)
Comment (jta) [reply .github/workflows/tf-account.yaml:28]: tailscale connects you to a network, you still have to provide the egress address.

File: .github/workflows/tf-account.yaml (hunk 1 above)
Comment (jta) [reply .github/workflows/tf-account.yaml:75]: That was my first take, but if we only want to proxy traffic intended for the Observe control plane you end up having quite a long list in `NO_PROXY`.

**Review by bendrucker — APPROVED** [review]
Forgot to press submit earlier 🤦🏻‍♂️

File: .github/workflows/tf-account.yaml [hunk 4]
```diff
@@ -37,19 +50,34 @@ jobs:
         with:
           terraform_version: ${{ inputs.terraform-version }}
 
+      - name: Tailscale
+        uses: tailscale/github-action@v2
+        # a tailscale oauth client requires tags
+        if: inputs.tailscale-tags != ''
+        with:
+          oauth-client-id: ${{ secrets.TS_OAUTH_CLIENT_ID }}
```
[the remark this answers is not in the document]
Comment (jta) [reply .github/workflows/tf-account.yaml:58]: Leaving as is for now since it matches all the tailscale docs.

File: .github/workflows/tf-account.yaml [hunk 5]
```diff
@@ -37,19 +50,34 @@ jobs:
...
+          tags: ${{ inputs.tailscale-tags }}
+
       - name: Terraform Format
         id: fmt
         run: terraform fmt -check
 
       - name: Configure AWS Credentials
         uses: aws-actions/configure-aws-credentials@v2
+        env:
+          HTTPS_PROXY: ${{ inputs.https-proxy }}
```
[the remark this answers is not in the document]
Comment (jta) [reply .github/workflows/tf-account.yaml]: Ack.

File: .github/workflows/tf-account.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (bendrucker) [reply .github/workflows/tf-account.yaml:58]: I don't think they made a considered choice. I'm anti-abbreviations because of the impact on discoverability. If you Google this identifier without double quotes you get a bunch of TypeScript stuff.

Their own docs aren't even consistent:

https://tailscale.com/kb/1276/tailscale-github-action/

File: .github/workflows/tf-account.yaml [hunk 6]
```diff
@@ -123,4 +151,7 @@ jobs:
 
       - name: Terraform Apply
         if: github.ref == format('refs/heads/{0}', github.event.repository.default_branch)
+        env:
```
Comment (bendrucker) [inline .github/workflows/tf-account.yaml:154]: Leftover?

**Review by bendrucker — APPROVED** [review]

**Review by bendrucker — APPROVED** [review]

**Review by bendrucker — APPROVED** [review]
